pkg/gardenlet/controller/federatedseed/network-policy/namespace_reconciler.go

```diff
@@ -0,0 +1,104 @@
+// Copyright (c) 2020 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package network_policy
+
+import (
+	"context"
+
+	corev1listers "k8s.io/client-go/listers/core/v1"
+
+	"github.com/gardener/gardener/pkg/gardenlet/controller/federatedseed/network-policy/helper"
+
+	"github.com/sirupsen/logrus"
+	corev1 "k8s.io/api/core/v1"
+	apiequality "k8s.io/apimachinery/pkg/api/equality"
+	kubecorev1listers "k8s.io/client-go/listers/core/v1"
+	kubenetworkingv1listers "k8s.io/client-go/listers/networking/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+// namespaceReconciler implements the reconcile.Reconcile interface for namespace reconciliation.
+type namespaceReconciler struct {
+	ctx                 context.Context
+	log                 *logrus.Entry
+	seedClient          client.Client
+	seedName            string
+	endpointsLister     kubecorev1listers.EndpointsLister
+	networkPolicyLister kubenetworkingv1listers.NetworkPolicyLister
+	namespaceLister     corev1listers.NamespaceLister
+}
+
+// newNamespaceReconciler returns the new namespace reconciler.
+func newNamespaceReconciler(ctx context.Context, seedLogger *logrus.Entry, seedClient client.Client, endpointsLister kubecorev1listers.EndpointsLister, networkPolicyLister kubenetworkingv1listers.NetworkPolicyLister, namespaceLister corev1listers.NamespaceLister, seedName string) reconcile.Reconciler {
+	return &namespaceReconciler{
+		ctx:                 ctx,
+		seedClient:          seedClient,
+		endpointsLister:     endpointsLister,
+		networkPolicyLister: networkPolicyLister,
+		namespaceLister:     namespaceLister,
+		log:                 seedLogger,
+		seedName:            seedName,
+	}
+}
+
+// Reconcile reconciles namespace in order to create the "allowed-to-seed-apiserver" Network Policy
+func (r *namespaceReconciler) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	namespace, err := r.namespaceLister.Get(request.Name)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+
+	if namespace.DeletionTimestamp != nil {
+		r.log.Infof("Do not update NetworkPolicy %q in namespace %q - namespace has a deletion timestamp", helper.AllowToSeedAPIServer, err)
```

Comment:
Debug log level might be better as there might be lots of namespaces in the seed with this condition.